### PR TITLE
Rlechanges

### PR DESCRIPTION
**Pull request by Vikram9866**

### CODE REVIEW

File: jpegenc/subblocks/RLE/RLECore/entropycoder.py [hunk 1]
```diff
@@ -8,31 +8,31 @@ def two2bin(num):
     return inum + 1
 
 
-def nbits(num, width):
+def nbits(num, width_data):
```
Comment (cfelton) [inline jpegenc/subblocks/RLE/RLECore/entropycoder.py:11]: I don't understand this function?  What does this do different than `int.bit_length()` or `log(num, 2)`?  
(there might be something I just am not currently understanding?).

Why does the fall through case `return num[width_data]`, doesn't make sense to me?

File: jpegenc/subblocks/RLE/RLECore/entropycoder.py [hunk 2]
```diff
@@ -8,31 +8,31 @@ def two2bin(num):
...
         if num[i] == 1:
             return i + 1
         i = i - 1
-    return num[width]
+    return num[width_data]
 
 
 @block
-def entropycoder(width, clock, reset, data_in, size, amplitude):
+def entropycoder(width_data, clock, reset, data_in, size, amplitude):
```
Comment (cfelton) [inline jpegenc/subblocks/RLE/RLECore/entropycoder.py:22]: I don't think you need `width_data` and a parameter to this function (the parameters should come after the ports), or is width_data a port?  Is `width_data` not `len(data_in)`?

Comment (Vikram9866) [inline jpegenc/subblocks/RLE/RLECore/entropycoder.py:22]: Yes, that's been changed .. forgot to add new file

File: jpegenc/subblocks/RLE/RLECore/entropycoder.py (hunk 1 above)
Comment (Vikram9866) [inline jpegenc/subblocks/RLE/RLECore/entropycoder.py:11]: Yes. int.bit_length() can be used .. But in conversion it shows me an error...

```
    Unsupported attribute: bit_length

```

Comment (cfelton) [inline jpegenc/subblocks/RLE/RLECore/entropycoder.py:11]: Understand now, you needed a convertible version, I still don't understand the last return though?

Comment (Vikram9866) [inline jpegenc/subblocks/RLE/RLECore/entropycoder.py:11]: The last return is for number 0 .. 
It do not enter the while loop and return the data from the last return

**Pull request closed without merging**